iothub_client/src/iothub_client_core_ll.c

```diff
@@ -2767,10 +2872,173 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_LL_UploadMultipleBlocksToBlobEx(IOTHUB_CLI
     }
     else
     {
-        result = IoTHubClient_LL_UploadMultipleBlocksToBlob_Impl(iotHubClientHandle->uploadToBlobHandle, destinationFileName, getDataCallbackEx, context);
+        char* uploadCorrelationId;
+        char* azureBlobSasUri;
+
+        if (IoTHubClient_LL_UploadToBlob_InitializeUpload(
+                iotHubClientHandle->uploadToBlobHandle, destinationFileName, &uploadCorrelationId, &azureBlobSasUri) != IOTHUB_CLIENT_OK)
+        {
+            LogError("Failed initializing upload in IoT Hub");
+            result = IOTHUB_CLIENT_ERROR;
+        }
+        else
+        {
+            bool uploadSucceeded;
+            IOTHUB_CLIENT_LL_UPLOADTOBLOB_CONTEXT_HANDLE uploadContext = IoTHubClient_LL_UploadToBlob_CreateContext(iotHubClientHandle->uploadToBlobHandle, azureBlobSasUri);
+
+            if (uploadContext == NULL)
+            {
+                LogError("Failed creating upload to blob context");
+                uploadSucceeded = false;
+            }
+            else
+            {
+                if (IoTHubClient_LL_UploadToBlob_UploadMultipleBlocks(uploadContext, getDataCallbackEx, context) != IOTHUB_CLIENT_OK)
+                {
+                    LogError("Failed to upload multiple blocks to Azure blob");
+                    uploadSucceeded = false;
+                }
+                else
+                {
+                    uploadSucceeded = false;
```

Comment:
false???

Reply:
Yeah, that was already fixed. Thanks.

Reply:
Btw we have an unit test for that, so it's safe.